src/tag_parsers.rs

```diff
@@ -190,13 +190,90 @@ pub fn parse_61_tag(field: &Field) -> Result<StatementLine, ParseError> {
+                .as_str()
+                .to_string();
+            let mut structured = InformationToAccountOwner::Structured {
+                transaction_code,
+                posting_text: None,
+                prima_nota: None,
+                unknown11: None,
+                purpose: None,
+                applicant_bin: None,
+                applicant_iban: None,
+                applicant_name: None,
+                return_debit_notes: None,
+                recipient_name: None,
+                unknown38: None,
+                additional_purpose: None,
+                unknown70: None,
+            };
+            if let InformationToAccountOwner::Structured {
+                ref mut posting_text,
+                ref mut prima_nota,
+                ref mut unknown11,
+                ref mut purpose,
+                ref mut applicant_bin,
+                ref mut applicant_iban,
+                ref mut applicant_name,
+                ref mut return_debit_notes,
+                ref mut recipient_name,
+                ref mut unknown38,
+                ref mut additional_purpose,
+                ref mut unknown70,
+                ..
+            } = structured
+            {
+                for sf in inner {
+                    let mut sf_inner = sf.into_inner();
+                    let tag = sf_inner.next().unwrap().as_str();
+                    let value = sf_inner.next().unwrap().as_str().to_string();
+                    match tag {
+                        "00" => *posting_text = Some(value),
+                        "10" => *prima_nota = Some(value),
+                        "11" => unknown11.get_or_insert_with(String::new).push_str(&value),
+                        "20" | "21" | "22" | "23" | "24" | "25" | "26" | "27" | "28" | "29" => {
+                            purpose.get_or_insert_with(String::new).push_str(&value)
+                        }
+                        "30" => *applicant_bin = Some(value),
+                        "31" => *applicant_iban = Some(value),
+                        "32" | "33" => applicant_name
+                            .get_or_insert_with(String::new)
+                            .push_str(&value),
+                        "34" => *return_debit_notes = Some(value),
+                        "35" => *recipient_name = Some(value),
+                        "38" => unknown38.get_or_insert_with(String::new).push_str(&value),
+                        "60" | "61" | "62" => additional_purpose
+                            .get_or_insert_with(String::new)
+                            .push_str(&value),
+                        "70" | "71" | "72" | "73" | "74" | "75" | "76" | "77" | "78" | "79" => {
+                            unknown70.get_or_insert_with(String::new).push_str(&value)
+                        }
+                        tag => {
+                            panic!("Unknown tag {} with value {}", tag, value)
```

Comment:
I don't think it's cool to panic here. Why don't we return an error here rather? All of the plumbing for that should already be in place.